app/controllers/SecureMessagingController.scala

```diff
@@ -106,4 +108,10 @@ class SecureMessagingController @Inject() (
 
   private def defineUploadLink(refererUrl: String) =
     routes.MrnEntryController.onPageLoad(Some(RedirectUrl(refererUrl))).url
+
+  private def defineH1Text(partialBody: String) = {
+    val h1textPattern: Regex = "(?i)>(.*?)<\\/h1>".r
+    val matcher = h1textPattern.findFirstIn(partialBody).getOrElse("");
+    StringUtils.substringBetween(matcher, ">", "</h1>")
```

Comment:
You don't need the `StringUtils.substringBetween` if you use the group function of the regex Matcher object, like this:
`val maybeMatcher = h1textPattern.findFirstMatchIn(partialBody)
    maybeMatcher.map{ _.group(1)}.getOrElse("")`